admin/atlas_client.go

```diff
@@ -55,27 +57,35 @@ func UseDigestAuth(apiKey, apiSecret string) ClientModifier {
 
 // UseOAuthAuth provides OAuthAuth authentication for Go SDK.
 // Method is provided as helper to create a default HTTP client that supports OAuth (Service Accounts) authentication.
-// credentials.LocalTokenCache can be supplied to reuse OAuth Token across application restarts.
-// Warning: for advanced use cases please use credentials.NewTokenSource directly in your code pass it to UseHTTPClient method.
 // Warning: any previously set httpClient will be overwritten. To fully customize HttpClient use UseHTTPClient method.
-func UseOAuthAuth(clientID, clientSecret string, tokenCache credentials.LocalTokenCache) ClientModifier {
+func UseOAuthAuth(clientID, clientSecret string) ClientModifier {
 	return func(c *Configuration) error {
-		var tokenSource credentials.TokenSource
-		if tokenCache != nil {
-			tokenSource = credentials.NewTokenSourceWithOptions(credentials.AtlasTokenSourceOptions{
-				ClientID:     clientID,
-				ClientSecret: clientSecret,
-				TokenCache:   tokenCache,
-				BaseURL:      &c.Servers[0].URL,
+		oauth := credentials.NewConfig(clientID, clientSecret)
+		ctx := context.WithValue(context.Background(), oauth2.HTTPClient,
+			&http.Client{
+				Transport: credentials.NewOAuthCacheTransport(http.DefaultTransport,
+					&c.UserAgent, nil),
 			})
-		} else {
-			tokenSource = credentials.NewTokenSourceWithOptions(credentials.AtlasTokenSourceOptions{
-				ClientID:     clientID,
-				ClientSecret: clientSecret,
-				BaseURL:      &c.Servers[0].URL,
+		source := oauth.TokenSource(ctx)
+		httpClient := oauth2.NewClient(ctx, source)
+		c.HTTPClient = httpClient
+		return nil
+	}
+}
+
+// UseOAuthAuthWithCache provides OAuthAuth authentication for Go SDK with the ability to reuse cached OAuth Token
+// Method is provided as helper to create a default HTTP client that supports OAuth (Service Accounts) authentication.
+// Warning: any previously set httpClient will be overwritten. To fully customize HttpClient use UseHTTPClient method.
```

Comment:
just becuase is in a new line doesn't mean it renders in a new line 
```suggestion
//
// Warning: any previously set httpClient will be overwritten. To fully customize HttpClient use UseHTTPClient method.
```